### ISSUE TITLE
Update C++ examples in folder /examples

### ISSUE THREAD

**Issue by Adda0** [VeriFIT/mata#161]
The examples in `/examples` are not automatically tested. Hence, when Mata changes its interface, the examples are left unmodified and many of them do not compile as of now.

We should fix all the examples and optionally modify the automatic tests to build and test the examples as well.

We need to:
- [x] Fix the current examples.
- [ ] Decide what kind of examples to write. What should they show?
- [ ] Automate the process of running all examples with each build, but do not block merging a PR if they fail.

We want examples for:
- iterating over transitions
- reversing NFA
- printing NFA

**Comment by martinhruska** [comment]
I agree that it should be part of automatical build. It always bothers me when I check and example and it is not compilable because of changed interface. There is no hope to keep it up to date without manual tests.

**Comment by Adda0** [comment]
It was decided that we will fix these examples. However, we will merge some simpler examples into one extended example showing basic use of Mata. The advanced examples will remain as they are.

Furthermore, we will build and test these examples in our automatic tests.

**Comment by martinhruska** [comment]
And where will be the extended example placed?

**Comment by Adda0** [comment]
In the same folder `examples/`. Along with the Jupyter notebooks, more sophisticated C++ examples and others.

**Comment by Adda0** [comment]
The current examples are fixed as a by-product of #283.
